docs/usage/channels.rst

```diff
@@ -0,0 +1,452 @@
+
+    flowchart TD
+        Broker[(Broker)]
+
+        Broker --> Backend_1(Backend)
+        Broker --> Backend_2(Backend)
+
+        Backend_1 --> Plugin_1{{Plugin}}
+        Backend_2 --> Plugin_2{{Plugin}}
+
+        Plugin_1 --> Subscriber_1[Subscriber]
+        Plugin_1 --> Subscriber_2[Subscriber]
+        Plugin_1 --> Subscriber_3[Subscriber]
+
+        Plugin_2 --> Subscriber_4[Subscriber]
+        Plugin_2 --> Subscriber_5[Subscriber]
+        Plugin_2 --> Subscriber_6[Subscriber]
+
+
+The ``ChannelsPlugin``
+----------------------
+
+.. currentmodule:: litestar.channels.plugin
+
+The :class:`ChannelsPlugin` acts as the central entity for managing channels and
+subscribers. It's used to publish messages, control how data is stored, manage
+subscribers, route handlers and configuration.
+
+
+.. tip::
+    The plugin makes itself available as a dependency under the ``channels`` key, which
+    means it's not necessary to import it and instead, it can be used from within route
+    handlers or other callables within the dependency tree directly
+
+
+Configuring the channels
++++++++++++++++++++++++++
+
+The channels manged by the plugin can be either defined upfront, passing them to the
+``channels`` argument, or created "on the fly" (i.e. on the first subscription to a
+channel) by setting ``arbitrary_channels_allowed=True``.
+
+
+.. code-block:: python
+    :caption: Passing channels explicitly
+
+    from litestar.channels import ChannelsPlugin
+
+    channels_plugin = ChannelsPlugin(..., channels=["foo", "bar"])
+
+
+.. code-block:: python
+    :caption: Allowing arbitrary channels
+
+    from litestar.channels import ChannelsPlugin
+
+    channels_plugin = ChannelsPlugin(..., arbitrary_channels_allowed=True)
+
+
+If ``arbitrary_channels_allowed`` is not ``True``, trying to publish or subscribe to a
+channel not passed to ``channels`` will raise a :exc:`ChannelsException`.
+
+
+Publishing data
++++++++++++++++
+
+
+One of the core aspects of the plugin is publishing data, which is done through its
+:meth:`publish <ChannelsPlugin.publish>` method:
+
+.. code-block:: python
+
+    channels.publish({"message": "Hello"}, "general")
+
+
+The above example will publish the data to the channel ``general``, subsequently putting
+it into all subscriber's :term:`event stream` to be consumed.
+
+This method is non-blocking, even though channels and the associated
+:term:`backends <backend>` are fundamentally asynchronous.
+
+Calling ``publish`` effectively enqueues a message to be sent to the backend, from which
+follows that there's no guarantee that an event will be available in the backend
+immediately after this call.
+
+Alternatively, the asynchronous :meth:`wait_published <ChannelsPlugin.wait_published>`
+method can be used, which  the internal message queue, publishing the data to the
+backend directly.
+
+.. note::
+    While calling :meth:`publish <ChannelsPlugin.publish>` does not guarantee the
```

Comment:
```suggestion
    Calling :meth:`publish <ChannelsPlugin.publish>` does not guarantee the
```

Reply:
I don't like this change in combination with the next one. It makes the sentence sound forced and chopped.